### ISSUE TITLE
Issues with dark theme on KDE

### ISSUE THREAD

**Issue by mgunyho** [manns/pyspread#144]
A new file in pyspread under KDE Plasma 5.9 with the Breeze Dark theme looks like this:
![image](https://user-images.githubusercontent.com/20118130/28502764-455ee156-7002-11e7-9baf-793977e2d35e.png)
There are a number of problems:
- Ugly gradient on the toolbar background (this I can live with, LibreOffice had a similar issue for a while), not visible on the screenshot.
- The grid color is black which makes it almost invisible (also black on the border selection combo box)
- The cursor showing which cell is highlighted is also black. When moving the cursor with the arrow keys either left or up, a white version of it blinks in the cell where the cursor was previously. It doesn't happen when going right or down. When moving the cursor by clicking, the same thing happens but sometimes the white cursor stays there until moving it again.
- When editing a cell, the white background makes the text hard to read.
- The font choice combo box has a black font

I've looked into the grid color thing a little bit. The problem seems to be that `wx.SystemSettings.GetColour` returns a black color for `wx.SYS_COLOUR_GRAYTEXT` so the issue is probably with how wx, GTK and KDE interface with each other so I'm not sure if there's much to do there. I can try to figure out some of the other stuff as well, although I'm not very experienced with wx or GUIs in general.

**Comment by manns** [comment]
Yes Breeze Dark is pretty broken when you use it with wxPython. Everything including Window borders are defined as black. Therefore, the grid is rendered in an unusable way. If you told me, which system color is right for all themes then I could easily fix this issue.

However, this seems a moving target. The only thing that I see feasible is to disable theme support for the grid as well as for some widgets. I hesitate doing that because with most themes, it works nicely.

As a quick fix, I recommend that you select all cells and redefine the text color and the background color. However, Breeze dark still breaks part of the GUI.

**Comment by mgunyho** [comment]
Yeah, not much can be done with the grid color. I guess graytext is the most sensible system color for it and the problem is with Breeze Dark not setting it properly (with regular Breeze it works correctly I think). 

I fixed some of these, see the commit. The cell background color was probably the biggest issue for usability.

**Comment by manns** [comment]
@mgunyho: Thank you for your pull request. This improves dark themes a lot.

I am reviewing all gtk2 themes on Debian right now. Most themes seem to work well.

The following themes show issues:

- Adwaita-Dark: No grid lines on default
- Albatross: Row and column labels are almost black on black, font choice is almost black on black each 2nd line
- Blackbird: Row and column labels are almost black on black, font choice is almost black on black each 2nd line
- BlackMATE: Table choice widget almost black on black, menu very light
- Blue submarine: Row and column labels are almost black on black, font choice is almost black on black each 2nd line
- Breeze-Dark: No grid lines on default
- ContrastHighInverse:  No grid lines on default, some widgets are invisible (no borders, black on black background)
- GreenSubmarine: Row and column labels are almost black on black, Font choice is almost black on black each 2nd line
- Greybird: Chosen font in font choice dialog is almost unreadable on blue background
- shearwater:  Row and column labels are almost black on black, Font choice is almost black on black each 2nd line
- Xfce-dusk: No grid lines on default

Could you please try tweaking colors further in order to maximize themeability?

**Comment by mgunyho** [comment]
Yeah, I can try to look into it this week. Hopefully it's not too hard to do without breaking the light themes or the Windows theme or something.

I guess some other sysvalue should be used for the grid lines if graytext is invisible in so many themes.

Not sure if it's a part of some of those you mentioned, but one thing I also noticed is that the border of the column labels, drawn with `wx.mixins.gridlabelrenderer.GridlabelRenderer.Drawborder()`, isn't easily adjustable so theming that will probably require a bit more work.

**Comment by manns** [comment]
I agree that if the grid labels are to be changed they need to be drawn manually. Ideally, this would be done with the same cairo based renderer functions that are used for drawing the grid.

I am going to look into this when you provide me with the system colors for the label backgrounds, grid lines and texts.

**Comment by mgunyho** [comment]
Setting the grid color to `wx.SYS_COLOUR_WINDOWTEXT` seems to work for most themes. I thought that it might make seeing which cell is being edited harder but it looks fine. I can try to fix some of the remaining issues as well.

**Comment by manns** [comment]
@mgunyho: Please let me know when you want me to review the changes.

**Comment by mgunyho** [comment]
I fixed the chart dialog tabs. I had to use a gradient as that seems like the only way to set the color of the divider between the tabs and panel content. I hope setting the colors is done at the correct point in the code.

CSV import: I tried commenting out these lines in `gui/_gui_dialogs.py` (~145-152):
```
dc.SetBackgroundMode(wx.SOLID)

if is_selected:
    dc.SetBrush(wx.Brush(wx.BLUE, wx.SOLID))
    dc.SetPen(wx.Pen(wx.BLUE, 1, wx.SOLID))
else:
    dc.SetBrush(wx.Brush(wx.WHITE, wx.SOLID))
    dc.SetPen(wx.Pen(wx.WHITE, 1, wx.SOLID))
```
It didn't fix the issue with BlackMATE, but at least with Breeze-Dark the combo boxes look more correct, except the background flashes white when (de)focusing. Overall I think that the cell background ordinal is broken for BlackMATE which leads to this issue and there isn't much to do about it.

The font choice combo box issue with Greybird seems to have disappeared for me. It might be because I cleared pyspreadrc.

Sorry about the messy commits again, I'm pretty new to working with branches.
